server/shared/src/main/scala/org/http4s/server/middleware/ResponseLogger.scala

```diff
@@ -31,121 +32,165 @@ import cats.~>
 import fs2.Chunk
 import fs2.Pipe
 import fs2.Stream
+import org.http4s.server.middleware.ResponseLogger.Lift
 import org.log4s.getLogger
 import org.typelevel.ci.CIString
 
+sealed abstract class ResponseLogger[F[_]] {
```

Comment:
Probably we can name it builder or factory (I prefer builder if someone would wonder 😆)? Also, shouldn't we hide it from public API?

Reply:
> Probably we can name it builder or factory (I prefer builder if someone would wonder 😆)? 

I think `ResponseLoggerBuilder` implies that the build action generates a `ResponseLogger`, but it doesn't - it transforms an `Http` instance to add logging. Given this class already has an `apply` method that adds logging to a client, I think it's fair to describe it as a logging middleware rather than a builder for one.

> Also, shouldn't we hide it from public API?

I don't think so - the idea is that users will call the `.withXXX` methods directly, so that we don't need to keep adding more factory methods to handle different configuration possibilities